frontend/src/modules/products/components/PreviewModal.tsx

```diff
@@ -0,0 +1,176 @@
+          Xem trước
+        </Modal.Title>
+      </Modal.Header>
+      <Form onSubmit={handleSubmit}>
+        <Modal.Body className="d-flex flex-column gap-4">
+          <Form.Group controlId="formProvince">
+            <Form.Control
+              className="form-select"
+              as="select"
+              required
+              value={selectedProvince}
+              onChange={e => {
+                setSelectedProvince(e.target.value);
+                setFormData({...formData, provinceCode: e.target.value});
+              }}
+            >
+              <option value="" disabled>
+                Tỉnh, thành phố
+              </option>
+              {formProvinces.map((province, index) => (
+                <option key={index} value={province.code}>
+                  {province.name}
+                </option>
+              ))}
+            </Form.Control>
+          </Form.Group>
+          <Form.Group controlId="formDistrict">
+            <Form.Control
+              className="form-select"
+              as="select"
+              required
+              value={selectedDistrict}
+              onChange={e => {
+                setSelectedDistrict(e.target.value);
+                setFormData({...formData, districtCode: e.target.value});
+              }}
+            >
+              <option value="" disabled>
+                Quận, huyện, thị xã
+              </option>
+              {districts?.map((district, index) => (
+                <option key={index} value={district.code}>
+                  {district.name}
+                </option>
+              ))}
+            </Form.Control>
+          </Form.Group>
+          <Form.Group controlId="formWard">
+            <Form.Control
+              className="form-select"
+              as="select"
+              required
+              value={formData.wardCode}
+              onChange={e =>
+                setFormData({...formData, wardCode: e.target.value})
+              }
+            >
+              <option value="" disabled>
+                Phường, xã, thị trấn
+              </option>
+              {wards?.map((ward, index) => (
+                <option key={index} value={ward.code}>
+                  {ward.name}
+                </option>
+              ))}
+            </Form.Control>
+          </Form.Group>
+          <Form.Group controlId="formDetailedAddress">
+            <Form.Control
+              type="text"
+              required
+              maxLength={100}
+              placeholder="Địa chỉ chi tiết"
+              onChange={e =>
+                setFormData({...formData, addressDetail: e.target.value})
+              }
+            />
+            <Form.Control.Feedback type="invalid">
+              Vui lòng nhập địa chỉ chi tiết
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Modal.Body>
+        <Modal.Footer>
+          {/* <AppButton variant="secondary" onClick={hideModal}>*/}
```

Comment:
why comment out of this section? if not use, remove it, Github already save our life